File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.java
import java.io.*;
import java.util.*;

public class Main {
    static ArrayList<Integer>[] A;
    static boolean visited[];

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        A = new ArrayList[n + 1];
        visited = new boolean[n + 1];
        for (int i = 1; i < n + 1; i++) {
            A[i] = new ArrayList<Integer>();
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int s = Integer.parseInt(st.nextToken());
            int e = Integer.parseInt(st.nextToken());
            A[s].add(e);
            A[e].add(s);
        }
        int cnt = 0;
        for (int i = 1; i < n + 1; i++) {
            if (!visited[i]) {
                cnt++;
                DFS(i);
            }
        }

        System.out.println(cnt);

    }

    private static void DFS(int i) {
        if (visited[i]) {
            return;
        }

        visited[i] = true;
        for (int t : A[i]) {
            if (visited[t] == false) {
                DFS(t);
            }
        }
    }
}

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.swift
let NM = readLine()!.split { $0 == " " }.map { Int($0)! }
    let (N, M) = (NM[0], NM[1])

    var graph = [[Int]](repeating: [], count: N + 1)
    var visited = [Bool](repeating: false, count: N + 1)

    for _ in 0 ..< M {
        let UV = readLine()!.split { $0 == " " }.map { Int($0)! }
        let (U, V) = (UV[0], UV[1])
        graph[U].append(V)
        graph[V].append(U)
    }
    
    func dfs(_ node: Int) {
        visited[node] = true
        for next in graph[node] {
            if !visited[next] {
                dfs(next)
            }
        }
    }
    
    var components = 0
    for i in 1 ... N {
        if !visited[i] {
            dfs(i)
            components += 1
        }
    }
    print(components)
